### DISCUSSION TITLE
Thanks for the face recognition

### DISCUSSION THREAD

**Discussion by jean-louis67** [photoprism/photoprism#1590] [Show and tell]
Hello

After doing the photoprism update including the face recognition feature, I re-indexed  this weekend the 10,000 photos in my library. And after…. great! I understood that the software would identify faces in the photos but not that there would be also comparisons. For example, in the case of a member of my family, from a dozen faces at different ages, I find more than 300 photos! It’s really exciting and I don’t think I’ve covered it all will bring.

This or these algorithms are truly remarkable. Are these algorithms specific to photoprism? In any case what is really appreciable is the ease of use of the functionality: naming faces, rejecting errors is done in a very intuitive way.
In some photos, for example when there are several people, some faces are not detected. Perhaps we could one day consider the possibility of helping the algorithm by circumscribing an area?

I end this message with a little humor. Here is a 'face' extracted by photoprism from a patchwork photo. It's a beautiful smiley!
![image](https://user-images.githubusercontent.com/65403912/135800661-918f2634-3431-424f-8576-caa1d56ce060.png)

Jean-Louis

**Comment by srett** [comment]
I'm also impressed by how well the recognition and clustering works. It speeds up categorizing faces a lot.  
I'm coming from digiKam, so I'm used to a different workflow, where you first have to train the algorithm for every single person by marking a dozen detected faces of each person individually, and then the recognition of that person improves for consecutive runs. It's quite tedious.  
However, the face detection of digiKam is much better, albeit slower. PhotoPrism is missing a lot of faces for me. Over the next couple days I'll try to figure out how I can manually inject a face region into the PP database in such a way that PP considers it a new face region that it hasn't run recognition on yet, so I can import all face regions from digiKam and then let PP just do the recognition step on these. I hope that's possible. :->

**Comment by graciousgrey** [comment]
Thanks for your feedback. This really keeps us motivated :) 

@jean-louis67 manual tagging is something we plan for the future (see https://github.com/photoprism/photoprism/issues/1548). Regarding the algorithms. We use a library for face detection. The recognition @lastzero build on his own based on some research papers.

@srett We would love to import face region data from other tools, if possible. We have a open ticket to import adobes xmp face region data (https://github.com/photoprism/photoprism/issues/554). Feel free to open another one for digikam :) . Maybe they use a similar format?

**Comment by hasdf** [comment]
Recognition-Algorithm is truly amazing. Saves a lot of time for photojournalists who are looking for the perfect picture of a person.

Will manual face-tagging (#1548) have an effect on the detection algorithm? I have 2-3 people who seem to have such weird faces, that Pigo doesn't want to detect them 🤣
